src/solvers/dg_unstructured_quad/dg_2d.jl

```diff
@@ -276,56 +215,48 @@ end
     primary_side    = element_side_id[boundary]
 
-    for primary_index in eachnode(dg)
-      # hacky way to set "exact solution" boundary conditions. Only used to test the orientation
-      # for a mesh with flipped elements
-      u_external = initial_condition((node_coordinates[1, primary_index, boundary],
-                                      node_coordinates[2, primary_index, boundary]),
-                                      t, equations)
-
-      # pull the left state from the boundary u values on the primary element as well as the
-      # directional vectors and scaling
-      #   Note! this assumes a conforming approximation, more must be done in terms of the normals
-      #         for hanging nodes and other non-conforming approximation spaces
-      u_ll = get_one_sided_surface_node_vars(u, equations, dg, 1, primary_index, boundary)
+    # pull the external state function from the bounary condition dictionary
+    external_state = boundary_condition[ name[boundary] ]
 
-      normal_vector  = get_surface_normal(normals, primary_index, primary_side, primary_element)
-      scaling_ll  = scaling[primary_index, primary_side, primary_element]
+    for primary_index in eachnode(dg)
+      # get the external solution values from the prescribed external state
+      #  TODO: make this more general and able to also depend on the normal direction
+      u_external = external_state((node_coordinates[1, primary_index, boundary],
+                                   node_coordinates[2, primary_index, boundary]), t, equations)
 
-      # rotate states
-      u_tilde_ll       = rotate_to_x(u_ll, normal_vector, equations)
-      u_tilde_external = rotate_to_x(u_external, normal_vector, equations)
+      # pull the left state from the boundary u values on the primary element
+      u_ll = get_one_sided_surface_node_vars(u, equations, dg, 1, primary_index, boundary)
 
-      # Call pointwise Riemann solver in the rotated direction
-      flux_tilde = surface_flux(u_tilde_ll, u_tilde_external, 1, equations)
+      # pull the outward pointing (normal) directional vector
+      outward_direction = get_surface_normal(normal_directions, primary_index, primary_side,
+                                             primary_element)
 
-      # backrotate the flux into the original direction
-      flux = rotate_from_x(flux_tilde, normal_vector, equations)
+      # Call pointwise numerical flux function in the rotated direction on the boundary
+      #    Note! the direction is normalized inside this function
+      flux = surface_flux(u_ll, u_external, outward_direction, equations)
 
-      # Scale the flux appropriately and copy back to primary element storage
+      # Copy flux back to primary element storage
       for v in eachvariable(equations)
-        surface_flux_values[v, primary_index, primary_side, primary_element] = flux[v] * scaling_ll
+        surface_flux_values[v, primary_index, primary_side, primary_element] = flux[v]
       end
     end
```

Comment:
I think this should get the same kind of interface we're using for the other solvers/meshes (setting the boundary flux instead of the boundary state, e.g. via the general Dirichlet BC).

Reply:
That is my overall plan. I am incrementally working towards and trying to workout the best strategy of where such a structure should "live" in the ecosystem.

Reply:
The boundary flux computation now uses an interface called `calc_boundary_flux_unstructured!` that is similar to the other solvers/meshes

Reply:
Thanks :+1: